### ISSUE TITLE
Audio & Video out of Sync

### ISSUE THREAD

**Issue by sumocomputers** [wouterdebie/locast2dvr#28]
I am using locast2dvr, and it works really well.

I am having an issue with recorded content where the audio and video are out of sync a little.

I understand this could be a Plex issue with mpeg2, but I don't have similar issues using a HDHomeRun Quatro and antenna.

I have tried recording different channels and even different zip codes, resulting in the same issue.

As for locast2dvr, I am running it pretty stock, and I have tried ffpmeg, but not direct stream yet.

Wondered if there are any known issues or other things I can try.

These are the only options I am enabling:

override_zipcodes="90001"
direct="0"
verbose="1"

SERVER
Mac mini 2018 i5 16GB RAM
locast2dvr v0.8.1
ffpmeg v4.3.2
Plex Media Server v1.22.0.4163

CLIENTS
Amazon Fire TV Cube 4K v8.13.2
Roku Ultra 4K v6.7.15 & v6.6.8 (Beta)
Apple TV 1080p 4th Gen v7.14.1
iOS v7.14.1

**Comment by wouterdebie** [comment]
This shouldn't be a locast2dvr problem, since it doesn't do any
manipulation of the MPEG stream. You could try the direct mode to remove
ffmpeg from the equation, since it might remux audio and video.
That said, I have seen this before on the actual locast stream. It might be
that locast doesn't sync A/V correctly. Have you tried the same stream from
locast on their website?

**Comment by Croq360** [comment]
I've noticed that Locast audio and video are slightly out of sync too on stuff I have watched back, though it's very slight, and I haven't reported on it.  I've assumed that this is a Locast problem, not a Locast2DVR problem.  Note that I'm using direct mode, not ffmpeg.  I'll watch some live TV today, both in Plex and through Locast app directly, and will report whether I see the same problem there or not.

**Comment by wouterdebie** [comment]
I noticed the same a while ago, but not for all channels. I assume that
it's on the locast side unfortunately. When using direct, the only thing
that locast2dvr does is copy over the stream and I assume ffmpeg doesn't do
anything else either.

**Comment by Croq360** [comment]
Yeah, if I recall correctly, when I looked at the ffmpeg options being used, it's doing a copy, no extra processing of audio or video.  That being said, I did have other issues with ffmpeg with missing blocks of the show, it seemed that it wound up falling behind such that blocks would get missed in the Locast data - that's part of the reason wouterdebie added the direct mode, and that works SO much better.  Note that my ffmpeg experience might not be common, just reporting on what happened for me.

**Comment by sumocomputers** [comment]
This is all great info, and looks like I have some work to do with gathering some data.

I had been avoiding watching directly on Locast because I had to go through a lot of VPN hoops to get Locast to think my location was valid.

**Comment by Croq360** [comment]
All the "direct" option does is to take ffmpeg out of the loop.  Locast2DVR handles everything in that mode.  I don't think it changes anything on the Locast side of things.

**Comment by wouterdebie** [comment]
@crock360 is correct. The API calls to locast are exactly the same, but
instead of ffmpeg handling the eventual stream, locast2dvr does that. I
have actually considered removing ffmpeg support form locast2dvr, since
there doesn't seem to be a good reason for keeping it around. However, I'm
still working on locast2tuner (the much faster Rust implementation) and I
have ditched the feature there.

**Comment by sumocomputers** [comment]
So after a bunch of experimentation, looks like I had the perfect storm of troubleshooting hell.

Turns out the shows I was having issues with seem to be problematic even in other locations (tried Cleveland and San Francisco in addition to Los Angeles).

Jeopardy was the big problem, and it doesn't matter whether it is carried by ABC or NBC, the audio is always out of sync.

I haven't watched directly on Locast because I always seem to screw up my valid location, but I am pretty sure it is a Locast issue.

I have also turned on direct mode permanently for the reasons mentioned by others.

I also spent a lot of time trying to get the one line terminal command to run at startup on macOS, and ended up writing an Apple Script that I exported to a .app file, and have that run at startup. I wondered if there is any way to share this with other users? I couldn't get any other method found online to work correctly (Automator, shell scripts, etc.)

Also, it appears the location override tricks Locast into accepting the location as valid, regardless of where the machine running locast2dvr is actually located. It would be good to let users know this more explicitly, assuming my understanding is correct.

Just want to say thank you again for such a great program.
